server/chat-plugins/chatlog.ts

```diff
@@ -518,49 +528,71 @@ const LogSearcher = new class {
 		}
 		return {results, total};
 	}
-
-	async ripgrepSearch(roomid: RoomID, search: string, limit?: number | null) {
-		let output;
-		if (!limit || limit > MAX_RESULTS) limit = MAX_RESULTS;
+	async ripgrepSearchMonth(roomid: RoomID, search: string, limit: number, date: string) {
+		let results;
 		try {
-			const options = [
-				'-e', `[^a-zA-Z0-9]${search.split('').join('[^a-zA-Z0-9]*')}([^a-zA-Z0-9]|\\z)`,
-				`${__dirname}/../../logs/chat/${roomid}`,
+			const {stdout} = await execFile('rg', [
+				'-e', this.constructRegex(search),
+				`logs/chat/${roomid}${toID(date) === 'all' ? '' : `/${date}`}`,
 				'-C', '3',
 				'-m', `${limit}`,
-			];
-			output = await execFile('rg', options, {maxBuffer: Infinity, cwd: path.normalize(`${__dirname}/../`)});
-		} catch (error) {
-			if (error.message.includes('Command failed')) return LogViewer.error(`No results found.`);
-			return LogViewer.error(`${error.message}`);
-		}
-		return this.render(
-			output.stdout.split('--').reverse(),
-			roomid,
-			search,
-			limit
-		);
+				'-P',
+			], {
+				maxBuffer: Infinity,
```

Comment:
`maxBuffer: Infinity` sounds like you _want_ out-of-memory errors?

Reply:
Wasn't sure how this should be handled, i don't know enough about this particular setting in order to figure out the best way to set it.

Reply:
Should this just be using ``process.memoryUsage().heapTotal``?

Reply:
No.

Reply:
I would probably set it to some const.

Reply:
Is there a good way to determine said const, though?

Reply:
Apparently `maxBuffer` defaults to 1MB. Feel free to raise that to, like, 64MB if you need it. I'd be wary of raising it higher than that.

Reply:
Also, I think limiting results to 100-200 a day should be plenty.

Reply:
Maybe tell people to redo the search in a specific day if they want more than 200 results in a day? Then cap it at 201.

Reply:
That seems annoying for users, tbh. 
Why not just cap it at 64 and then lower ``MAX_RESULTS`` a bit?

Reply:
200 seems inconvenient for active rooms, so i capped it at 64mb per month and made it catch and return if it's a maxBuffer exceeded error.

Reply:
Crashing the entire server seems much more annoying for users.

I suspect if you think it's annoying, it's because you don't understand what I'm recommending.

Reply:
well, yes, lol, crashing would be worse.
From what i understand, you're recommending 200 results per day, with more only if it's specifically for that day?